flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -3190,4 +3190,36 @@ def fir_CUDADataTransferOp : fir_Op<"cuda_data_transfer", []> {
   }];
 }
 
+def fir_CUDAAllocateOp : fir_Op<"cuda_allocate", [AttrSizedOperandSegments,
+    MemoryEffects<[MemAlloc<DefaultResource>]>]> {
```

Comment:
Should it also have a write effect on the box reference operand?

Reply:
Yes that would make sense. I'll add it

Reply:
Done. I added also effect on `errmsg`, `pinned` and `source`. `errmsg`, `pinned` are diagnostic variable that will be written to if present and `source` will be used for source allocation so we need to read from it.